### ISSUE TITLE
Missing scanf family of functions

### ISSUE THREAD

**Issue by stefanct** [embeddedartistry/libc#175]
If there is no intention to add support for them, I think it should at least be documented as it is quite essential for parsing data in some applications. What's your stance on the issue? Is there any suggested workaround currently?

**Comment by phillipjohnston** [comment]
I have no stance on these functions :). I haven't developed a system that has required them, and you are the first to ask. No technical or ideological reason.

**Comment by phillipjohnston** [comment]
[picolibc (newlib-nano) has full support](https://github.com/picolibc/picolibc/blob/main/doc/printf.md). I only mention as an immediate solution - I assume you are here because you do not wish to use newlib.

**Comment by phillipjohnston** [comment]
Looking over these functions, one blocker is that I do not actually support `FILE` or `stdin/out/err` in this library, so it would be gated on having a suitable solution for that.

**Comment by stefanct** [comment]
Yes, in this regard it is similar to some of the `printf` functions. `sscanf` is what I was after in this case. It's not a big deal for me and I can work around it.
